src/tracing/swapFlowLoggers.ts

```diff
@@ -0,0 +1,33 @@
+import { SwapEventName } from '@uniswap/analytics-events'
+import { sendAnalyticsEvent } from 'analytics'
+
+import { SwapEventTimestampTracker, SwapEventType } from './SwapEventTimestampTracker'
+
+const tracker = SwapEventTimestampTracker.getInstance()
+
+export function logSwapSuccess(hash: string, chainId: number, analyticsContext: any) {
+  const hasSetSwapSuccess = tracker.hasTimestamp(SwapEventType.FIRST_SWAP)
+  const elapsedTime = tracker.setElapsedTime(SwapEventType.FIRST_SWAP)
+  sendAnalyticsEvent(SwapEventName.SWAP_TRANSACTION_COMPLETED, {
+    // We only log the time-to-swap metric for the first swap of a session,
+    // so if it was previously set we log undefined here.
+    time_to_swap: hasSetSwapSuccess ? undefined : elapsedTime,
+    time_to_swap_since_first_input: hasSetSwapSuccess
+      ? undefined
```

Comment:
qq: should we do something other than `undefined` for second+ swaps? This way we know when parsing logs that it's not because the variable was just never set? Maybe something like `-1`

Reply:
i get that but logging a placeholder number here seems more dangerous to cause an incorrect analysis in amplitude, if we forget to filter and they are quietly included in some aggregations